components/DetailedCharacter.js

```diff
@@ -0,0 +1,147 @@
+import useStore from '../hooks/useStore';
+import styled, { css } from 'styled-components';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import Image from 'next/image';
+
+export default function DetailedCharacterCard({ characterId }) {
+  const { push } = useRouter();
+
+  function handleClose(e) {
+    e.stopPropagation();
+    push('/characters');
+  }
+
+  const character = useStore(state =>
+    state.characters.find(character => characterId === character.id)
```

Comment:
you could create a function in your state called 'getCharacterById' this would increase the readability of your code.